syntax = "proto3";

package NullSpaceIPC;



message SimpleHaptic {
	repeated string regions = 1;
	uint32 effect = 2;
	float strength = 3;
	float duration = 4;
}

message RealtimeHaptic {
	message Magnitude {
		string region = 1;
		float strength = 2;
	}
	
	repeated Magnitude magnitudes = 1;

}


message PlaybackEvent {
	enum Command {
		UNKNOWN = 0;
		PAUSE = 1;
		UNPAUSE = 2;
		CANCEL = 3;
	}
	Command command = 1;
	
}
message HighLevelEvent {
	uint64 parent_id = 1;
	oneof events {
		PlaybackEvent playback_event = 2;
		SimpleHaptic simple_haptic = 3;
		RealtimeHaptic realtime_haptic = 4;
	}
}
